### PR TITLE
Fix the loss of replies

### PR DESCRIPTION
**Pull request by royrapoport**

### CODE REVIEW

File: message_downloader.py [hunk 1]
```diff
@@ -98,8 +97,10 @@ def download(self, start_at = None):
                             cid, message['thread_ts'])
                         thread_messages = [
                             x for x in thread_messages if x['thread_ts'] != x['ts']]
+                        message['replies'] = thread_messages
                         self.MessageWriter.write(
                             thread_messages, cid, parent_user_id)
+                self.MessageWriter.write(messages, cid)
```
Comment (jdq22) [inline message_downloader.py:103]: Can you share more about how this fixes the bug?

Comment (royrapoport) [reply message_downloader.py:103]: The way we keep track of the replies to a message is to have a field in the message called 'replies' which contains a list of replies to the message.  

Now, I'm not exactly sure what was different a month ago, but right now, the problem we had was that we would:
1. Get the list of messages for a channel (which would not include threaded messages); 
2. Save those messages to the database;
3. For each that were a threadhead, get the messages in the thread they started
4. Save those messages to the database

So we were accurately capturing all the messages, but we were not capturing the RELATIONSHIP between them.  

So instead, what we're doing now is 
1. Get the list of messages for a channel (which would not include threaded messages); 
2. For each that's a threadhead, get thread messages and save those in DB
3. Set the 'replies' field for each threadhead to the thread messages
4. Save the top-level messages (now with a 'replies' field) to the DB

File: user_and_channel_report [hunk 2]
```diff
@@ -12,6 +13,8 @@ import slack_channel_report
 import report_generator
 import report_targets
 
+roy = "U06NSQT34"
```
Comment (jdq22) [inline user_and_channel_report]: Can you move this into a file that's not committed to git?

Comment (royrapoport) [reply user_and_channel_report]: Yeah, just submitted a change.

**Review by jdq22 — APPROVED** [review]

**Pull request merged**